pyomo/contrib/doe/fim_doe.py

```diff
@@ -6,11 +6,84 @@
 import time
 import pickle
 from itertools import permutations, product
-from pyomo.contrib.sensitivity_toolbox.sens import sipopt, sensitivity_calculation
-#from idaes.apps.uncertainty_propagation.sens import get_dsdp
-from pyomo.contrib.sensitivity_toolbox.sens import get_dsdp
+from pyomo.contrib.sensitivity_toolbox.sens import sipopt, sensitivity_calculation, get_dsdp
 
-class DesignOfExperiments: 
+class Measurement_flatten:
+    def __init__(self, measurement_timeset, measurement_extra_index):
+        '''
+        This class stores measurements' information
+        Parameters:
+        ----------
+        measurement_timeset: a dictionary, keys are measurement names in the model, values are a list of time they are measured
+        measurement_extra_index: A dictionary, keys are the measurement names, values are a list of indexes. If no extra indexes, values are None.
+        '''
+        self.measurement_name = list(measurement_timeset.keys())
+        self.measurement_timeset = measurement_timeset
+        self.measurement_extra_index = measurement_extra_index
+        # which measurement has extra indexes
+        self.extra_measure_name = list(measurement_extra_index.keys())
+
```

Comment:
We should add logic here to test if the measurement object is valid. We should also add logic to test if a measurement object is a subset of another measurement object.